corehq/apps/smsforms/models.py

```diff
@@ -208,9 +231,34 @@ def sync_sql_session_from_couch_session(couch_session):
     except SQLXFormsSession.DoesNotExist:
         sql_session = SQLXFormsSession(couch_id=couch_id)
 
-    for attr, value in data.items():
-        setattr(sql_session, attr, value)
+    for attr in SESSION_PROPERTIES_TO_SYNC:
+        setattr(sql_session, attr, data.get(attr, None))
+
+    # hack to avoid excess saves. see sync_couch_session_from_couch_session
+    sql_session.do_not_sync = True
     sql_session.save()
+    sql_session.do_not_sync = False
+
+
+@receiver(post_save, sender=SQLXFormsSession)
+def sync_signal_catcher(sender, instance, *args, **kwargs):
+    sync_couch_session_from_couch_session(instance)
+
+
+def sync_couch_session_from_couch_session(sql_session):
```

Comment:
I think this should say sync_couch_session_from_sql_session, but I won't block the pr on it